### PR TITLE
Throw error for add_to_expression with GenericNonlinearExpr

### LINKED ISSUE #3505: Extend `add_to_expression!` for NonlinearExpr
Hi, it would be nice to use `add_to_expression!` with `NonlinearExpr`. The following currently errors on master:
```julia
(@v1.9) pkg> st JuMP
Status `~/.julia/environments/v1.9/Project.toml`
  [4076af6c] JuMP v1.14.1 `https://github.com/jump-dev/JuMP.jl.git#master`

using JuMP
m=Model(); @variable(m,x);
ex = zeros(NonlinearExpr)
add_to_expression!(ex, exp(x))
ERROR: MethodError: no method matching add_to_expression!(::Array{NonlinearExpr, 0}, ::NonlinearExpr)

Closest candidates are:
  add_to_expression!(::GenericQuadExpr{C, V}, ::Union{GenericAffExpr{C, V}, V}, ::Union{Number, LinearAlgebra.UniformScaling}) where {C, V}
   @ JuMP ~/.julia/packages/JuMP/8MuvO/src/quad_expr.jl:364
  add_to_expression!(::GenericQuadExpr{C, V}, ::V) where {C, V}
   @ JuMP ~/.julia/packages/JuMP/8MuvO/src/quad_expr.jl:323
  add_to_expression!(::GenericQuadExpr{C, V}, ::V, ::GenericAffExpr{C, V}) where {C, V}
   @ JuMP ~/.julia/packages/JuMP/8MuvO/src/quad_expr.jl:417
  ...


```

### PR DESCRIPTION
**Pull request by odow**
Closes #3505

We can always revisit this in future if more people ask for it.

### CODE REVIEW

File: test/test_nlp_expr.jl [hunk 1]
```diff
@@ -926,4 +926,20 @@ function test_generic_nonlinear_expr_infer_variable_type()
...
+    @variable(model, x)
+    y = zero(NonlinearExpr)
+    @test_throws(
+        ErrorException(
+            "`add_to_expression!` is not supported for expressions of type " *
+            "`$(typeof(y))` because they cannot be modified in-place. " *
+            "Instead of `add_to_expression!(expr, args..)`, use " *
+            "`expr += *(args...)`.",
+        ),
+        test_add_to_expression!(y, 2.0, sin(x)),
```
Comment (odow) [inline test/test_nlp_expr.jl:940]:
```suggestion
        add_to_expression!(y, 2.0, sin(x)),
```

**Comment by odow** [comment]
I don't think we can redirect to `add_mul!!` because `add_to_expression!` is in-place only.

**Comment by hdavid16** [comment]
That works. `add_mul!!` is behind the internal `_MA` in JuMP. So would the user need to load `MA` separately from `JuMP`?

**Comment by odow** [comment]
> So would the user need to load MA separately from JuMP?

Yes

**Comment by blegat** [comment]
I meant redirecting in the error message

File: test/test_nlp_expr.jl [hunk 2]
```diff
@@ -926,6 +926,29 @@ function test_generic_nonlinear_expr_infer_variable_type()
...
 
+function test_add_to_expression!()
+    model = Model()
+    @variable(model, x)
+    y = zero(NonlinearExpr)
+    @test_throws(
+        ErrorException(
+            """
+            `add_to_expression!` is not supported for expressions of type
+            `$(typeof(f))` because they cannot be modified in-place.
```
Comment (odow) [inline test/test_nlp_expr.jl:937]:
```suggestion
            `$(typeof(y))` because they cannot be modified in-place.
```

**Comment by odow** [comment]
Done

**Review by blegat — APPROVED** [review]

**Pull request merged**